### PR TITLE
server_respawn_lifecycleNode

### PR DESCRIPTION
**Pull request by iampaulidrobo**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | (add tickets here #2752 ) |
| Primary OS tested on | (Ubuntu 20.04) |
| Robotic platform tested on | (Turtlebot3 simulation) |

---

## Description of contribution in a few bullet points
*The lifecycle manager gets a heartbeat timeout and restarts all the nodes, making the system working again.

## Description of documentation updates required from your changes
1)All node set to respawn true ,to make node running again in case of failure.
2)Lifecycle manager node edited with checking for CONFIGURE and ACTIVATE, and do them in case they are not ACTIVATED.

---

## Future work that may be required in bullet points
1)Reverse transition for the lifecycle nodes to function with modularity.

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

File: nav2_bringup/bringup/launch/localization_launch.py [hunk 1]
```diff
@@ -101,7 +103,8 @@ def generate_launch_description():
             executable='lifecycle_manager',
             name='lifecycle_manager_localization',
             output='screen',
+            respawn=True,
             parameters=[{'use_sim_time': use_sim_time},
-                        {'autostart': autostart},
+                        {'autostart': True},
```
Comment (SteveMacenski) [inline nav2_bringup/bringup/launch/localization_launch.py:108]: Revert, autostart is a parameterization

File: nav2_bringup/bringup/launch/navigation_launch.py [hunk 2]
```diff
@@ -138,8 +143,9 @@ def generate_launch_description():
             executable='lifecycle_manager',
             name='lifecycle_manager_navigation',
             output='screen',
+            respawn=True,
             parameters=[{'use_sim_time': use_sim_time},
-                        {'autostart': autostart},
```
Comment (SteveMacenski) [inline nav2_bringup/bringup/launch/navigation_launch.py:142]: Revert

File: nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager.hpp [hunk 3]
```diff
@@ -30,130 +30,132 @@
...
-  // Whether to automatically start up the system
-  bool autostart_;
-
-  bool system_active_{false};
-};
-
-}  // namespace nav2_lifecycle_manager
-
-#endif  // NAV2_LIFECYCLE_MANAGER__LIFECYCLE_MANAGER_HPP_
+  class LifecycleManager : public rclcpp::Node
```
Comment (SteveMacenski) [inline nav2_lifecycle_manager/include/nav2_lifecycle_manager/lifecycle_manager.hpp:40]: Replace original formatting

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

**Comment by SteveMacenski** [comment]
This PR is critically flawed, we should discuss in the ticket. This does not in any way check or aware of the servers failing, its just naively just always trying to bring up, not only when a server was respawned due to a crash. This would cause a critical looping error if we wanted to bring down the lifecycle system safely.

**Comment by iampaulidrobo** [comment]
Yes,sure.
The server comes up whenever any one node goes down.
@SteveMacenski

**Pull request closed without merging**